bundle/regal/rules/performance/non-loop-expression/non_loop_expression_test.rego

```diff
@@ -0,0 +1,389 @@
+package regal.rules.performance["non-loop-expression_test"]
+
+import data.regal.ast
+import data.regal.config
+import data.regal.rules.performance["non-loop-expression"] as rule
+
+test_loop_start_points_walks_are_loops if {
+	sps := rule._loop_start_points with input as ast.policy(`
+allow if {
+	walk(foo, [path, value])
+	value == "foo"
+}
+`)
+
+	sps == {"0": {5: {
+		{"location": "5:13:5:17", "type": "var", "value": "path"},
+		{"location": "5:19:5:24", "type": "var", "value": "value"},
+	}}}
+}
+
+test_loop_start_points_some_k_v if {
+	sps := rule._loop_start_points with input as ast.policy(`
+allow if {
+	some k, v in input.foo
+	k == 1
+	some value in v
+	v == 2
+}
+`)
+
+	sps == {"0": {
+		5: {
```

Comment:
Interesting how the first level variable can't be numeric (due to that type checker bug), but seems to work for this next level. That should be fixed of course, but certainly out of scope for this PR :)

Reply:
I'm hoping that @johanfylling can fix that in OPA soon :)